dist_run.py

```diff
@@ -114,8 +119,6 @@ def main():
         model = Transformer(config)
 
     model.setup_caches(1, 4096)
-    # TODO: refine this .to once we start using fp8 for KV cache
-    model = model.to(model_dtype)
```

Comment:
Curious -- 
If set/get precision are only effective for KV cache, why removing this `model.to` line is okay?

Reply:
Or, are you saying that the purpose of this line was previously to cast the dtype of the cache **only**?

Reply:
correct - I was only using this line to handle the dtype of the caches.  The rest of the weights are all being set in the checkpoint loading.
I also want to note that this set_precision is not going to be a long term fix b/c it's setting *two* caches dtype - right now it sets both kvcache and rope cache, but longer term we will want fp8/fp6/fp4 kvcache and yet continue with bf16 rope.. so this is not a perm fix, but resolves things for now.